src/nupic/algorithms/Connections.cpp

```diff
@@ -318,6 +318,7 @@ Segment Connections::getSegment(CellIdx cell, SegmentIdx idx) const {
 }
 
 const vector<Synapse> &Connections::synapsesForSegment(Segment segment) const {
+  NTA_ASSERT(segment < segments_.size()) << "Segment out of bounds! " << segment;
```

Comment:
NTA_CHECK would be ideal, but this is a low-level, performance critical method, so we probably can't afford that luxury.